Resources/views/Layouts/macros.html.twig

```diff
@@ -38,14 +38,15 @@
     </div>
 {% endmacro %}
 
-{% macro stop_point_code(id, stopPoint, class, title) %}
+{% macro stop_point_code(id, stopPoint, class, title, type = null) %}
     <div id="{{ id }}"  class="{{ class }}">
         {% set blockTitle = title ? title : 'stop_point.block.code.title' %}
         <div class="title">{{blockTitle|trans({}, 'default')}}</div>
 
         <div class="content">{% spaceless %}
             {% if stopPoint %}
-                {{'stop_point.block.code.content'|trans({'%code%':stopPoint.getExternalCode()}, 'default')}}
+                {% set type = type is null or stopPoint.codes|code(type) is null ? 'external_code' : type %}
```

Comment:
To remove logical in view. I think it's better to check if type is null return 'external_code' in "StopPointExtension.php" getCode function.
What do you think ?